pkg/hooks/webhook.go

```diff
@@ -99,15 +146,32 @@ func (w *WebhookExecutor) Call(request interface{}, response interface{}) error
 		logging.Logger.V(6).Info("Webhook response", "type", w.hookType, "url", w.url, "body", rawResponse)
 	}
 
-	// Check status code.
-	if resp.StatusCode != http.StatusOK {
+	if cacheEntry != nil {
+		// According to https://datatracker.ietf.org/doc/html/rfc7232
+		// When 'If-None-Match' is present, backend should respond 304 when object has not changed
+		if resp.StatusCode == 304 {
+			// TODO: Find a way to deep copy from cacheEntry.Response to response and switch cacheEntry.Response to store decoded response
+			if err = k8sjson.Unmarshal(cacheEntry.Response, response); err != nil {
```

Comment:
suggest: Let us avoid error shadowing. Let us use `if err := ...` instead i.e. `:=` instead of `=`

Reply:
done